src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -45,6 +45,10 @@ def decorate(cls):
 
 
 def AuthRequired(enabled=True):
+    if not isinstance(enabled, bool):
+        raise TypeError('AuthRequired used incorrectly. '
```

Comment:
You should better clarify that the argument needs to be a boolean.

Reply:
If ``enabled`` is not a boolean, then i think nobody understands the error message ``You are likely missing parentheses!``. Instead i think something like ``The function parameter must be of type boolean!`` should be used.

Reply:
Saying `The function parameter must be of type boolean!` doesn't explain why it isn't of type boolean and one might not immediately understand that the decorator has been used incorrectly. That's something that would be left unexplained, instead it would be explained what the code checked, but that's something which can easily be looked up and might as well be in the traceback. Therefore, I prefer to tell the developers what exactly he/she did wrong so that they can immediately take action and fix the issue without having to understand the mechanics behind that issue and don't have to think about why this check has been added or why the parameter isn't of type boolean and hence raised the exception.

Reply:
I still do not think that the current message really helps to identify a problem and i still can't find the association/link between parentheses and a wrongly used declarator.

Reply:
Would it be helpful then if I added an example like this?:

Wrong: @AuthRequired
Right: @AuthRequired()

Reply:
The problem is due to the way python decorators work. Python decorators can have arguments or not. If the decorator does not have arguments, we can use it like `@mydecorator` without parenthesis, if the decorator has arguments (even if every argument has a default value) we need to use it like `@mydecorator()`, with parenthesis.

What @p-na is trying to achieve is to warn the user that it cannot use the `@AuthRequired` decorator without parenthesis, and I agree with @votdev that the error message is not very informative to the user. Perhaps we could say `"AuthRequired decorator requires parenthesis. Example: @AuthRequired()"`

But just to add another thing. I'm removing the `@AuthRequired` decorator in PR #22283 so this discussion is not actually that important :-)